ext/standard/string.c

```diff
@@ -1283,16 +1283,10 @@ PHP_FUNCTION(implode)
 		tmp_glue = NULL;
 		pieces = arg1;
 	} else {
-		if (Z_TYPE_P(arg1) == IS_ARRAY) {
-			glue = zval_get_tmp_string(arg2, &tmp_glue);
-			pieces = arg1;
-			php_error_docref(NULL, E_DEPRECATED,
-				"Passing glue string after array is deprecated. Swap the parameters");
-		} else if (Z_TYPE_P(arg2) == IS_ARRAY) {
+		if (Z_TYPE_P(arg1) == IS_STRING) {
```

Comment:
Hm, this changes the behavior: Previously it would cast anything (that isn't an array) to a string, now it strictly requires a string. Ideally it would follow zpp rules...